### ISSUE TITLE
change username error

### ISSUE THREAD

**Issue by peterennis** [deckgo/deckdeckgo#298]
Username peterennis
Changed to adaept
Received a red error message "...could not change user name..."
Logout
Login and try again.
Screen remains gray after trying submit again.
Looks like this:

![Capture683](https://user-images.githubusercontent.com/140737/64214201-4d461200-ce64-11e9-8160-b355b0cc8f89.png)

**Comment by peterpeterparker** [comment]
thx for the issue

@nmattia could you plz trace what happened?
and also reset the username for Peter's user?

**Comment by nmattia** [comment]
There were two issues:

* 2019-09-03T22:51:40.792Z: the PUT failed because the JWT token was expired
* 2019-09-03T22:54:26.073Z: the PUT made the user anonymous because we assumed the frontend would always send an `email` field. @peterpeterparker what do you want to do about this?

**Comment by peterpeterparker** [comment]
@nmattia regarding the first issue:

1. a promise is missing on the app side once the call had been performed, which could gives the use the feeling that the request didn't complete even if it is the case. I will solve that shortly with a PR

2. I don't think that the JWT token was expired. I just created a brand new user and tried to change the username. It didn't worked out and the `response` received from the API is kind of weird (`anonymous=true, username=null`?)

what do you think about it?

<img width="1191" alt="Capture d’écran 2019-09-05 à 13 38 51" src="https://user-images.githubusercontent.com/16886711/64339065-7345f080-cfe3-11e9-8274-d34230d1c314.png">

<img width="1191" alt="Capture d’écran 2019-09-05 à 13 39 00" src="https://user-images.githubusercontent.com/16886711/64339066-7345f080-cfe3-11e9-8163-0c5c75bfe328.png">

**Comment by peterpeterparker** [comment]
This issue has now been solved.

The username couldn't be changed because the JWT token was expired. As a resolve() was missing the user couldn't notice the error

Furthermore, as the email was not emitted in the PUT, the username was removed.

Both problem has now been solved.

Regarding @peterennis user account and name, I contacted you on the slack channel, let's clear that there

Thx for testing our beta and for reporting this 👍
